src/sagemaker/session.py

```diff
@@ -4352,6 +4352,30 @@ def update_endpoint(self, endpoint_name, endpoint_config_name, wait=True):
             self.wait_for_endpoint(endpoint_name)
         return endpoint_name
 
+    def is_inference_component_based_endpoint(self, endpoint_name):
+        """Returns 'True' if endpoint is inference-component-based, 'False' otherwise.
+
+        An endpoint is inference component based if and only if the associated endpoint config
+        has a role associated with it and no production variants with a ``ModelName`` field.
+
+        Args:
+            endpoint_name (str): Name of the Amazon SageMaker ``Endpoint`` to determine
+                if inference-component-based.
+        """
+        describe_endpoint_response = self.describe_endpoint(endpoint_name)
+        endpoint_config_name = describe_endpoint_response["EndpointConfigName"]
+        describe_endpoint_config_response = self.sagemaker_client.describe_endpoint_config(
+            EndpointConfigName=endpoint_config_name
+        )
+        production_variants = describe_endpoint_config_response.get("ProductionVariants", [])
+        execution_role_arn = describe_endpoint_config_response.get("ExecutionRoleArn")
+        if len(production_variants) == 0:
```

Comment:
can it be a IC endpoint with no variants?

Reply:
So far, zero variant is not allowed if this endpoint will be used to deploy ICs.